zebra-chain/src/transaction/serialize.rs

```diff
@@ -564,3 +567,21 @@ impl ZcashDeserialize for Transaction {
         }
     }
 }
+
+impl<T> ZcashDeserialize for Arc<T>
```

Comment:
@hdevalence do we want to implement this for `Arc`?

Reply:
I'm not sure, I don't see any reason not to but I'm also not sure what the case is where this is more ergonomic (presumably it exists @yaahc ?)

Reply:
yea, right now the code that serializes `Vec<Transaction>` breaks without this impl if we add an Arc inbetween, it just allows types behind `Arc`s to serialize the same way they would directly, and we already have this exact impl for `Vec<T: ZcashDeserialize>` so I don't think theres any particular downside to adding this impl, it's still constrained on the inner type implementing `ZCashDeserialize`

Reply:
Cool, sounds good! I forgot about the impl on `Vec<T: ZcashDeserialize>` 🙈